x/incentives/keeper/distribute.go

```diff
@@ -722,6 +733,54 @@ func (k Keeper) distributeInternal(
 				// which is bounded to an Int. So we can safely skip this.
 
 				amtIntBi.Quo(amtIntBi, lockSumTimesRemainingEpochsBi)
+
+				// Determine if the value to distribute is worth enough in minValueForDistr denom to be distributed.
+				if coin.Denom == minValueForDistr.Denom {
+					// If the denom is the same as the minValueForDistr param, no transformation is needed.
+					if amtInt.LT(minValueForDistr.Amount) {
+						continue
+					}
+				} else {
+					// If the denom is not the minValueForDistr denom, we need to transform the underlying to it.
+					// Check if the denom exists in the cached values
+					value, ok := minDistrValueCache.denomToMinValueMap[coin.Denom]
+					if !ok {
+						// Cache miss, figure out the value and add it to the cache
+						poolId, err := k.prk.GetPoolForDenomPairNoOrder(ctx, minValueForDistr.Denom, coin.Denom)
+						if err != nil {
+							// If the pool denom pair pool route does not exist in protorev, we add a zero value to cache to avoid
+							// querying the pool again.
+							minDistrValueCache.denomToMinValueMap[coin.Denom] = zeroInt
+							continue
+						}
+						swapModule, pool, err := k.pmk.GetPoolModuleAndPool(ctx, poolId)
+						if err != nil {
+							return nil, err
+						}
+
+						minTokenRequiredForDistr, err := swapModule.CalcOutAmtGivenIn(ctx, pool, minValueForDistr, coin.Denom, sdk.ZeroDec())
+						if err != nil {
+							return nil, err
+						}
+
+						// Add min token required for distribution to the cache
+						minDistrValueCache.denomToMinValueMap[coin.Denom] = minTokenRequiredForDistr.Amount
+
+						// Check if the value is worth enough in the token to be distributed.
+						if amtInt.LT(minTokenRequiredForDistr.Amount) {
+							// The value is not worth enough, continue
+							continue
+						}
+					} else {
+						// Cache hit, use the value
+
+						// Check if the underlying is worth enough in the token to be distributed.
+						if amtInt.LT(value) || value.IsZero() {
```

Comment:
We probably want to split these two and add a comment to make clear what the zero value represents.

```suggestion
						// This route does not exist in protorev so a zero value has been added when a cache miss occurred 
						if value.IsZero() {
							continue
						}
						// Check if the underlying is worth enough in the token to be distributed.
						if amtInt.LT(value) {
```

Reply:
Good suggestion